### ISSUE TITLE
'rewrite_polyfills= false' is not working when language_out is present and below ECMASCRIPT_2017

### ISSUE THREAD

**Issue by icy0307** [google/closure-compiler#4080]
`rewrite_polyfills= false` still inject polyfills when language_out is below ECMASCRIPT_2017 for async function
# step to reproduce

1. echo 'export async function test(){console.log('test');}'> my_program.js
2. npx google-closure-compiler --js=my_program.js --js_output_file=out.js --rewrite_polyfills=false --language_out=ECMASCRIPT_2015

**Comment by L3P3** [comment]
I agree that it should use the `Promise` implementation (hopefully) available.

**Comment by eduardoacskimlinks** [comment]
We are experiencing the same problem, affecting our production users on version `google-closure-compiler: 20240317.0.0.`.

May I ask why this flag is no longer working?
